### PR TITLE
TTP: Adds a TPP button to the payments hub

### LINKED ISSUE #8368: [Mobile Payments] Add TTP button to the payments hub (M2)
This button should be optionally visible only when TPP is available

Image

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #8368 

### Description

Adds TPP button to the payment hub. The click on the button does nothing so far

### Testing instructions

* If TPP is available on the device - notice that there is a new TPP row on the payments hub
* If TPP is not available on the device - notice that the new row is not visible. Make sure that it is not visible on the release build

### Images/gif

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by jostnes** [comment]
@kidinov thanks for making this change! Having TTP as an item on the menu will definitely make it easier to access it. 

I was testing out the build and it worked fine on multiple stores at first, but when I changed from dark mode to light, the Tap To Pay item is suddenly moved to the bottom of the list and when I reverted back to dark mode but the TTP item is still displayed as the last item: 

https://user-images.githubusercontent.com/17252150/220813861-8ae34e31-4484-4d84-bbf8-242bcae317b2.mp4

Not sure if it's just me but it would be great if someone tries to reproduce this as well.

**Comment by jostnes** [comment]
Just to add as a note, I found 2 other issues not related to this change while on this PR (couldn't find existing issues for them so created them as new issues), the two linked above ^

**Comment by kidinov** [comment]
@jostnes 

I could not reproduce this =(. I made some changes 0a24ee138f05d78bd824320f1ae1a456d44865e4 which may fix that. Could you please check if you can reproduce it now?

**Comment by jostnes** [comment]
> Could you please check if you can reproduce it now?

Tried it again after the latest commits, and can't reproduce it now (tried the same steps I tried earlier, switching between multiple stores with/without TTP and changing from dark/light modes). Think it looks good now!

**Comment by kidinov** [comment]
I got feedback from the design:
* Payment options -> Actions
* Tap To Pay moved to Actions

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/payments/cardreader/hub/CardReaderHubViewModelTest.kt [hunk 1]
```diff
@@ -1229,6 +1238,126 @@ class CardReaderHubViewModelTest : BaseUnitTest() {
             // THEN
             verify(cardReaderTracker).trackCashOnDeliveryLearnMoreTapped()
         }
+
+    @Test
+    fun `given tpp available, when view model started, then show tpp row`() = testBlocking {
```
Comment (AnirudhBhat) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/payments/cardreader/hub/CardReaderHubViewModelTest.kt]: 🔍 Nothing related to this PR - I've seen usage tpp for Tap To Pay. Should it be ttp instead [**T**ap **T**o **P**ay]?

This is very minor and please feel free to ignore this comment.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/hub/CardReaderHubViewModel.kt [hunk 2]
```diff
@@ -84,6 +87,23 @@ class CardReaderHubViewModel @Inject constructor(
...
+                isOnboardingComplete = false,
+                cashOnDeliveryItem = cashOnDeliveryState.value!!
+            ),
+            isLoading = true,
+            onboardingErrorAction = null,
+            learnMoreIppState = learnMoreIppState,
+        )
+    )
+
+    val viewStateData: LiveData<CardReaderHubViewState> = viewState
```
Comment (AnirudhBhat) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/hub/CardReaderHubViewModel.kt:102]: ❤️

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/payments/cardreader/hub/CardReaderHubViewModelTest.kt [hunk 3]
```diff
@@ -1229,6 +1238,126 @@ class CardReaderHubViewModelTest : BaseUnitTest() {
...
+
+        // WHEN
+        initViewModel()
+
+        // THEN
+        assertThat((viewModel.viewStateData.getOrAwaitValue()).rows).anyMatch {
+            it is NonToggleableListItem &&
+                it.icon == R.drawable.ic_baseline_contactless &&
+                it.label == UiString.UiStringRes(R.string.card_reader_tap_to_pay) &&
+                it.description == UiString.UiStringRes(R.string.card_reader_tap_to_pay_description) &&
```
Comment (AnirudhBhat) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/payments/cardreader/hub/CardReaderHubViewModelTest.kt:1257]: 🔍 nitpick: Now that we have added a description field for NonToggleable list items as well. We could probably write tests to verify that the description isn't being shown for any of the NonToggleable list items except for TTP. WDYT?

PS: Again, very minor comment. Please feel free to ignore this one if it's not worth it.

**Review by AnirudhBhat — APPROVED** [review]
Thanks for working on this @kidinov! I've tested the feature and it works as expected. I've left a couple of very minor comments that are non-blocking for the PR merge.

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/payments/cardreader/hub/CardReaderHubViewModelTest.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/payments/cardreader/hub/CardReaderHubViewModelTest.kt]: Thank you for noticing that. I used it all wrong for a while 🤦‍♂️

Fixed now 922d798283052307632395fda93036d705e4e462

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/payments/cardreader/hub/CardReaderHubViewModelTest.kt (hunk 3 above)
Comment (kidinov) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/payments/cardreader/hub/CardReaderHubViewModelTest.kt:1257]: Good point. Added a test on that 342a0d3c8df865e5c000536249975c71cb94c705

**Pull request merged**